test/geometries/linestring.cpp

```diff
@@ -0,0 +1,106 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+// Unit Test
+
+// Copyright (c) 2020 Digvijay Janartha, Hamirpur, India.
+
+// Parts of Boost.Geometry are redesigned from Geodan's Geographic Library
+// (geolib/GGL), copyright (c) 1995-2010 Geodan, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <iostream>
+
+#include <geometry_test_common.hpp>
+
+#include <boost/core/ignore_unused.hpp>
+#include <boost/geometry/algorithms/make.hpp>
+#include <boost/geometry/algorithms/append.hpp>
+#include <boost/geometry/geometries/point.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/linestring.hpp>
+#include <boost/geometry/geometries/concepts/linestring_concept.hpp>
+#include <boost/geometry/geometries/adapted/c_array.hpp>
+#include <boost/geometry/geometries/adapted/boost_tuple.hpp>
+#include <boost/geometry/io/dsv/write.hpp>
+
+#include <test_common/test_point.hpp>
+
+BOOST_GEOMETRY_REGISTER_C_ARRAY_CS(cs::cartesian)
+BOOST_GEOMETRY_REGISTER_BOOST_TUPLE_CS(cs::cartesian)
+
+
+template <typename P>
+bg::model::linestring<P> create_linestring()
+{   
+    bg::model::linestring<P> l1;
+    P p1;
+    bg::assign_values(p1, 1, 2, 3);
+    bg::append(l1, p1);
+    return l1;
+}
+
+template <typename L, typename T>
+void check_linestring(L& to_check, T x, T y, T z)
+{
+    BOOST_CHECK_EQUAL(bg::get<0>(to_check[0]), x);
+    BOOST_CHECK_EQUAL(bg::get<1>(to_check[0]), y);
+    BOOST_CHECK_EQUAL(bg::get<2>(to_check[0]), z);
+}
+
+template <typename P>
+void test_construction()
```

Comment:
This in fact only tests default construction. Since we are at the unit test, I was wondering about such test cases

- `test_default_constructor`
- `test_copy_constructor`
- `test_copy_assignment`

Reply:
> what's your opinion

Since it is unit tests, I will add `copy_constructor` and `copy_assignment` as well. I completely agree with you. Just one thing, adding so many assertions would make testing slow right? But not much I guess, what's the approximate complexity of these functions we use for testing in boost?

Reply:
> adding so many assertions would make testing slow right?

In our case the compilation time, due to code traits unrelated to use of assertions (e.g. meta-programming, header-only, etc.), is much much longer than run time of any of our tests, even if you stick hundreds of assertions.

> But not much I guess, what's the approximate complexity of these functions we use for testing in boost?

If you have `a == b`, complexity depends on type of the operands. If `a` and `b` are `int` it will be very different than if they are `std::vector<int>`, etc.

Performance is not our concern here, but correctness.

Reply:
> Performance is not our concern here, but correctness.

Yes, I agree. I'll add other functions as well in the next commit.